activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -269,6 +269,10 @@ def transform_values(*args, &block)
       dup.tap { |hash| hash.transform_values!(*args, &block) }
     end
 
+    def compact
+      select { |_, value| !value.nil? }
```

Comment:
For posterity, changing this originally correct solution to `dup.compact!` introduced a bug: https://github.com/rails/rails/pull/28138